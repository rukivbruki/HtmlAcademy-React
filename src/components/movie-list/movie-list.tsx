// Libs
import * as React from 'react';
import {connect} from "react-redux";
// Utils
import {FilmsType, FilmType} from '../../types';
import {getShownCardsNumber} from "../../reducer/films/selectors.js";
// Components
import MovieCard from "../movie-card/movie-card";

interface Props {
  activeItem?: FilmType;
  filteredFilms: FilmsType;
  filter?: string;
  onOpenCard: () => void;
  shownCardsNumber: number;
  setActiveItem: () => void;
  removeActiveItem: () => void;
  setTimer: () => void;
  removeTimer: () => void;
}

const MovieList = (props: Props) => {
  const {shownCardsNumber, activeItem, filteredFilms} = props;
  const shownFilms = filteredFilms.slice(0, shownCardsNumber);

  return (
    <div className="catalog__movies-list">
      {shownFilms
        .map((film) => {
          const {name, id} = film;
          return (
            <MovieCard
              key={id}
              active={name === activeItem.name}
              {...film}
              {...props}
            />
          );
        })}
    </div>
  );
};

const mapStateToProps = (state) => ({
  shownCardsNumber: getShownCardsNumber(state)
});

export {MovieList};
export default connect(mapStateToProps)(React.memo(MovieList));
