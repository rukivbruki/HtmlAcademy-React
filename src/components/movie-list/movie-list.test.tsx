import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter as Router} from 'react-router-dom';
import configureStore from "redux-mock-store";
import {MovieList} from './movie-list';

const mockStore = configureStore([]);

const noop = () => void 0;

const film = {
  backgroundColor: `red`,
  backgroundImage: `img`,
  description: `desc`,
  director: `dir`,
  genre: `genre`,
  id: 1,
  isFavorite: false,
  name: `film`,
  posterImage: `img`,
  previewImage: `img`,
  previewVideoLink: `video`,
  rating: 1,
  released: 1,
  runTime: 1,
  scoresCount: 1,
  starring: [`test`],
  videoLink: `video`
};

const films = [film];

const props = {
  filteredFilms: films,
  onOpenCard: noop,
  shownCardsNumber: 1,
  setActiveItem: noop,
  removeActiveItem: noop,
  setTimer: noop,
  removeTimer: noop,
  activeItem: film,
};

it(`MovieList renders correctly`, () => {
  const store = mockStore({
    films,
    filteredFilms: films
  });

  const tree = renderer
    .create(
        <Router>
          <Provider store={store}>
            <MovieList {...props}/>
          </Provider>
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
