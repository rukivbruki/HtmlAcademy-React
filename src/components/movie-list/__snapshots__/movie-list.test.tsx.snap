// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="film"
        height="175"
        src="img"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "height": "100%",
            "width": "100%",
          }
        }
      >
        film
      </a>
    </h3>
    <div
      className="player"
    >
      <video
        autoPlay={true}
        className="player__video"
        muted={true}
        poster="img"
        src="video"
      />
    </div>
  </article>
</div>
`;
